runtime/src/bank.rs

```diff
@@ -331,33 +331,49 @@ impl CachedExecutors {
         })
     }
 
-    fn put(&mut self, pubkey: &Pubkey, executor: Arc<dyn Executor>) {
-        let entry = if let Some(mut entry) = self.executors.remove(pubkey) {
-            entry.executor = executor;
-            entry
-        } else {
+    fn put(&mut self, executors: &[(&Pubkey, Arc<dyn Executor>)]) {
+        let new_executors: Option<Vec<_>> = executors
+            .iter()
+            .map(|(key, executor)| {
```

Comment:
Could this be `filter_map`, the return be `Vec<_>` and the test in the following block `if !new_executors.is_empty()` ?